pkg/controller/controller_instance_test.go

```diff
@@ -649,19 +654,24 @@ func TestReconcileInstanceAsynchronousNoOperation(t *testing.T) {
 	updatedInstance := assertUpdateStatus(t, actions[0], instance)
 	assertInstanceReadyFalse(t, updatedInstance)
 
-	// The item should've been added to the pollingQueue for later processing
-	if testController.pollingQueue.Len() != 1 {
-		t.Fatalf("Expected the asynchronous instance to end up in the polling queue")
-	}
-	item, _ := testController.pollingQueue.Get()
-	if item == nil {
-		t.Fatalf("Did not get back a key from polling queue")
-	}
-	key := item.(string)
-	expectedKey := fmt.Sprintf("%s/%s", instance.Namespace, instance.Name)
-	if key != expectedKey {
-		t.Fatalf("got key as %q expected %q", key, expectedKey)
-	}
+	// Since polling is rate-limited, it is not possible to check whether the
+	// instance is in the polling queue.
+	//
+	// TODO: add a way to peek into rate-limited adds that are still pending,
+	// then uncomment.
+
+	// if testController.pollingQueue.Len() != 1 {
+	// 	t.Fatalf("Expected the asynchronous instance to end up in the polling queue")
+	// }
+	// item, _ := testController.pollingQueue.Get()
+	// if item == nil {
+	// 	t.Fatalf("Did not get back a key from polling queue")
+	// }
+	// key := item.(string)
+	// expectedKey := fmt.Sprintf("%s/%s", instance.Namespace, instance.Name)
+	// if key != expectedKey {
+	// 	t.Fatalf("got key as %q expected %q", key, expectedKey)
+	// }
```

Comment:
same here RE removing this or add an issue to re-add it to the test in a later PR